language/move-ir-compiler/ir-to-bytecode/src/labels.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use codespan_reporting::{
+    diagnostic::{Diagnostic, Label},
+    files::SimpleFiles,
+    term::{
+        emit,
+        termcolor::{ColorChoice, StandardStream},
+        Config,
+    },
+};
+use move_ir_types::{ast::*, location::*};
+use std::collections::HashSet;
+
+/// Verify that, within a single function, no two blocks use the same label, and all jump statements
+/// specify a destination label that exists on some block. If any block labels or statements don't
+/// meet these conditions, print diagnostics and return an error.
+pub fn verify_function(source: &str, function: &Spanned<Function_>) -> Result<()> {
```

Comment:
Yeah, I guess it depends... if by "generate code" you mean placing this logic in `ir-to-bytecode/src/compiler.rs`, such as in the `compile_function` function, I chose not to because AFAICT that means that this label verification would run on block labels generated by the move-lang compiler. My understanding is that move-lang HLIR generates labels with sequential numbers, so it could perform some aspects of this verification more efficiently. So, as-is, I wasn't sure if you'd want this to run in a critical path of the main compiler of the project. I could move it to `compiler.rs` (or someplace else if that's not what you had in mind) -- what do you think?

Reply:
Totally fine the verification runs on the compiler generated code. All of the other checks are :) 

So I think it would be fine to inline this check inside of `ir-to-bytecode/src/compiler.rs` if it makes things easier for you

Reply:
Also no need to go crazy on error message generation unless you want to :)